src/http/status.cr

```diff
@@ -0,0 +1,172 @@
+  def informational?
+    100 <= value <= 199
+  end
+
+  # Returns `true` if the response status code is between 200 and 299.
+  def success?
+    200 <= value <= 299
+  end
+
+  # Returns `true` if the response status code is between 300 and 399.
+  def redirection?
+    300 <= value <= 399
+  end
+
+  # Returns `true` if the response status code is between 400 and 499.
+  def client_error?
+    400 <= value <= 499
+  end
+
+  # Returns `true` if the response status code is between 500 and 599.
+  def server_error?
+    500 <= value <= 599
+  end
+
+  # Returns the default status description of the given HTTP status code.
+  def description : String?
+    case value
+    when 100 then "Continue"
+    when 101 then "Switching Protocols"
+    when 102 then "Processing"
+    when 200 then "OK"
+    when 201 then "Created"
+    when 202 then "Accepted"
+    when 203 then "Non-Authoritative Information"
+    when 204 then "No Content"
+    when 205 then "Reset Content"
+    when 206 then "Partial Content"
+    when 207 then "Multi-Status"
+    when 208 then "Already Reported"
+    when 226 then "IM Used"
+    when 300 then "Multiple Choices"
+    when 301 then "Moved Permanently"
+    when 302 then "Found"
+    when 303 then "See Other"
+    when 304 then "Not Modified"
+    when 305 then "Use Proxy"
+    when 306 then "Switch Proxy"
+    when 307 then "Temporary Redirect"
+    when 308 then "Permanent Redirect"
+    when 400 then "Bad Request"
+    when 401 then "Unauthorized"
+    when 402 then "Payment Required"
+    when 403 then "Forbidden"
+    when 404 then "Not Found"
+    when 405 then "Method Not Allowed"
+    when 406 then "Not Acceptable"
+    when 407 then "Proxy Authentication Required"
+    when 408 then "Request Timeout"
+    when 409 then "Conflict"
+    when 410 then "Gone"
+    when 411 then "Length Required"
+    when 412 then "Precondition Failed"
+    when 413 then "Payload Too Large"
+    when 414 then "URI Too Long"
+    when 415 then "Unsupported Media Type"
+    when 416 then "Range Not Satisfiable"
+    when 417 then "Expectation Failed"
+    when 418 then "I'm a teapot"
+    when 421 then "Misdirected Request"
+    when 422 then "Unprocessable Entity"
+    when 423 then "Locked"
+    when 424 then "Failed Dependency"
+    when 426 then "Upgrade Required"
+    when 428 then "Precondition Required"
+    when 429 then "Too Many Requests"
+    when 431 then "Request Header Fields Too Large"
+    when 451 then "Unavailable For Legal Reasons"
+    when 500 then "Internal Server Error"
+    when 501 then "Not Implemented"
+    when 502 then "Bad Gateway"
+    when 503 then "Service Unavailable"
+    when 504 then "Gateway Timeout"
+    when 505 then "HTTP Version Not Supported"
+    when 506 then "Variant Also Negotiates"
+    when 507 then "Insufficient Storage"
+    when 508 then "Loop Detected"
+    when 510 then "Not Extended"
+    when 511 then "Network Authentication Required"
+    else          nil
```

Comment:
I know this has been discussed, but I would change this to an empty string. Imagine having to pass the description to some place that expects a string, then you won't be able to use this method and you'll have to use `not_nil!` (which is probably incorrect, because you  don't want it to fail like that) or use `|| ""` which is what this method should return.

Reply:
Basically, if we don't know the description because the code isn't officially registered we shouldn't signal that with something noteworthy as `nil`, which requires additional handling: an empty string is fine and easier to use.

Reply:
or `"Unknown http code"` ?

Reply:
@bew The status message is supposed to explain the meaning of the code to human readers. Stating that it's unknown, doesn't explain anything about the status code, only about the processing handler. Unknown codes should either return `nil` or empty string which expresses that there is no explanation.

Reply:
I agree with @asterite about `nil` - if we can avoid an Union and a nil, that's all good :)

Reply:
ok, thank you @straight-shoota

Reply:
One could always do `description.to_s` which would return an empty string on `nil`.

Reply:
Good point. Lets leave nil then.

Reply:
Golang implementation makes it `"status code #{code}"` which is not very helpful IMO.